sdks/java/core/src/main/java/org/apache/beam/sdk/io/TFRecordIO.java

```diff
@@ -712,6 +713,12 @@ private TFRecordReader(TFRecordSource source) {
       }
 
       @Override
+      public boolean allowsDynamicSplitting() {
```

Comment:
Do not think so -- unit tests caught this issue.

I changed FileBasedSource in `FILEPATTERN` mode to be always splittable (since it is) even for globs that happen to be of a single file with no wildcards. In this case, that splitting itself will return a single file that is unsplittable.

I think this is one of many places where sources were correct because of that somewhat-strange behavior.

Adding this code to the reader itself seems safer and more straightforward.